File: PerfectNumber.py
import math
a = input()
a = int(a)
Uoc = []
Value = 0
for i in range(1,a,1):
    if (a % i == 0):
        Uoc.append(i)
for item in Uoc:
    Value += item
if (Value == a):
    print ("PerfectNumber")
else:
    print ("UnPerfectNumber")

File: ConsoleApp1/HomeWork17.04.25/PerfectNumber.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HomeWork17._04._25
{
    internal class PerfectNumber
    {
        static void Main()
        {
            int a = Convert.ToInt32(Console.ReadLine());
            int[] Uoc = new int[a];
            int count = 0;
            int Value = 0;
            for (int i = 1; i < a; i++)
            {
                if (a % i == 0)
                {
                    Uoc[count] = i;
                    count++;
                }
            }
            foreach (int item in Uoc)
            {
                Value += item;
            }
            Console.WriteLine(Value == a ? "SHH" : "Không HH");
        }
    }
}
